pkg/keyvalue/leveldb.go

```diff
@@ -71,6 +71,7 @@ func (b *batch) addToCache(cache *freecache.Cache) {
 
 func (b *batch) leveldbBatch() *leveldb.Batch {
 	b.mu.Lock()
+	defer b.mu.Unlock()
```

Comment:
`defer` adds some overhead on each function call. There is only one `return`, so using `defer` here looks unnecessary

Reply:
Yep, mostly I agree with your concerns, but this place is very "hot". So even a small overhead here can decrease node's performance.